core/src/main/java/com/yahoo/imapnio/async/request/AbstractFetchCommand.java

```diff
@@ -96,37 +136,58 @@ protected AbstractFetchCommand(final boolean isUid, @Nonnull final String msgNum
      * @param isUid whether prepending UID
      * @param msgNumbers the message numbers string
      * @param macro the macro
+     * @param partialExtUidFetchInfo partial uid fetch info
      */
-    protected AbstractFetchCommand(final boolean isUid, @Nonnull final String msgNumbers, @Nonnull final FetchMacro macro) {
+    protected AbstractFetchCommand(final boolean isUid, @Nonnull final String msgNumbers, @Nonnull final FetchMacro macro,
+                                   @Nullable final PartialExtensionUidFetchInfo partialExtUidFetchInfo) {
         this.isUid = isUid;
         this.msgNumbers = msgNumbers;
         this.macro = macro;
         this.dataItems = null;
+        this.partialExtUidFetchInfo = partialExtUidFetchInfo;
     }
 
     @Override
     public void cleanup() {
         this.msgNumbers = null;
         this.dataItems = null;
         this.macro = null;
+        this.partialExtUidFetchInfo = null;
     }
 
     @Override
+    @Nonnull
     public ByteBuf getCommandLineBytes() {
-        final ByteBuf sb = Unpooled.buffer();
-        sb.writeBytes(isUid ? UID_FETCH_SP_B : FETCH_SP_B);
-        sb.writeBytes(msgNumbers.getBytes(StandardCharsets.US_ASCII));
-        sb.writeByte(ImapClientConstants.SPACE);
+        final ByteBuf bb = Unpooled.buffer();
+        bb.writeBytes(isUid ? UID_FETCH_SP_B : FETCH_SP_B);
+        bb.writeBytes(msgNumbers.getBytes(StandardCharsets.US_ASCII));
+        bb.writeByte(ImapClientConstants.SPACE);
 
         if (dataItems != null) {
-            sb.writeByte(ImapClientConstants.L_PAREN);
-            sb.writeBytes(dataItems.getBytes(StandardCharsets.US_ASCII));
-            sb.writeByte(ImapClientConstants.R_PAREN);
+            bb.writeByte(ImapClientConstants.L_PAREN);
+            bb.writeBytes(dataItems.getBytes(StandardCharsets.US_ASCII));
+            bb.writeByte(ImapClientConstants.R_PAREN);
         } else {
-            sb.writeBytes(macro.name().getBytes(StandardCharsets.US_ASCII));
+            bb.writeBytes(macro.name().getBytes(StandardCharsets.US_ASCII));
+        }
+
+        if (isUid && partialExtUidFetchInfo != null) {
+            bb.writeByte(ImapClientConstants.SPACE);
+            bb.writeByte(ImapClientConstants.L_PAREN);
+            bb.writeBytes(PARTIAL_EXTENSION_SP_B);
+            final StringBuilder sb;
+            if (partialExtUidFetchInfo.getRange() == PartialExtensionUidFetchInfo.Range.LAST) {
+                sb = new StringBuilder().append(ImapClientConstants.MINUS).append(partialExtUidFetchInfo.getLowestUid())
```

Comment:
why do you create StringBuilder? you can directly do bb.writeBytes(...) to save memory.